allennlp/nn/util.py

```diff
@@ -1325,3 +1326,37 @@ def uncombine_initial_dims(tensor: torch.Tensor, original_size: torch.Size) -> t
     else:
         view_args = list(original_size) + [tensor.size(-1)]
         return tensor.view(*view_args)
+
+
+def inspect_parameters(module: torch.nn.Module, quiet: bool = False) -> Dict[str, Any]:
+    """
+    Inspects the model/module parameters and their tunability. The output is structured
+    in a nested dict so that parameters in same sub-modules are grouped together.
+    This can be helpful to setup module path based regex, for example in initializer.
+    It prints it by default (optional) and returns the inspection dict. Eg. output::
+
+        {
+            "_text_field_embedder": {
+                "token_embedder_tokens": {
+                    "_projection": {
+                        "bias": "parameter:tunable",
+                        "weight": "parameter:tunable"
+                    },
+                    "weight": "parameter:frozen"
```

Comment:
it seems like this docstring is not right? doesn't it just produce e.g. `"bias": "tunable"`?

Reply:
Oops! updated it.